modules/aggregation-layers/src/utils/gpu-grid-aggregation/gpu-grid-aggregator.js

```diff
@@ -223,16 +208,11 @@ export default class GPUGridAggregator {
   run(opts = {}) {
     // reset results
     this.setState({results: {}});
-    const aggregationParams = this.getAggregationParams(opts);
-    this.updateGridSize(aggregationParams);
-    const {useGPU} = aggregationParams;
-    if (this._hasGPUSupport && useGPU) {
-      return this.runAggregationOnGPU(aggregationParams);
-    }
-    if (useGPU) {
-      log.warn('GPUGridAggregator: GPU Aggregation not supported, falling back to CPU')();
+    const aggregationParams = this._normalizeAggregationParams(opts);
+    if (!this._hasGPUSupport) {
+      log.log(1, 'GPUGridAggregator: not supported')();
     }
-    return this.runAggregationOnCPU(aggregationParams);
+    return this.runAggregationOnGPU(aggregationParams);
```

Comment:
If CPU aggregation is removed from this class, it's no longer necessary to differentiate by the method name?